scripts/plot_trees.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Plots all trees from an MCMC run.
+
+Example Usage:
+poetry run python ../scripts/plot_trees.py
+
+Example Usage with arguments:
+    poetry run python scripts/plot_trees.py
+    --out_dir data/plots/tree/mark00
+    --data_fp data/mcmc/mark00/samples.json
+
+"""
+
+import argparse
+import logging
+
+
+from pathlib import Path
+from tqdm import tqdm
+
+import pyggdrasil.serialize as serialize
+import pyggdrasil.tree_inference as tree_inf
+import pyggdrasil.visualize as visualize
+
+
+def create_parser() -> argparse.Namespace:
+    """
+    Parser for required input user.
+
+    Returns:
+        args: argparse.Namespace
+    """
+
+    parser = argparse.ArgumentParser(description="Plot trees from MCMC sample.")
+
+    parser.add_argument(
+        "--out_dir",
+        required=True,
+        help="Output directory to save the tree plots.",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--data_fp",
+        required=True,
+        help="File path containing the mcmc samples.",
+        type=str,
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def main() -> None:
+    """
+    Main function for plotting trees from MCMC samples.
+
+    Returns:
+        None
+    """
+
+    args = create_parser()
+
+    # get the full path
+    out_dir = Path(args.out_dir)
+    # if out_dir does not exist, create it
+    if not out_dir.exists():
```

Comment:
I don't really get why the `if` statement is needed here – wouldn't `exist_ok=False` work here?

Reply:
Good point! Made a few changes to be more efficient here.